src/ui_ng/src/app/group/group.component.ts

```diff
@@ -0,0 +1,166 @@
+import { SessionService } from "./../shared/session.service";
+import { TranslateService } from "@ngx-translate/core";
+import { Observable } from "rxjs/Observable";
+import { Subscription } from "rxjs/Subscription";
+import { Component, OnInit, ViewChild, OnDestroy } from "@angular/core";
+import {operateChanges, OperateInfo, OperationService, OperationState} from "harbor-ui";
+
+import {
+  ConfirmationTargets,
+  ConfirmationState,
+  ConfirmationButtons
+} from "../shared/shared.const";
+import { ConfirmationMessage } from "../shared/confirmation-dialog/confirmation-message";
+import { ConfirmationDialogService } from "./../shared/confirmation-dialog/confirmation-dialog.service";
+import { AddGroupModalComponent } from "./add-group-modal/add-group-modal.component";
+import { UserGroup } from "./group";
+import { GroupService } from "./group.service";
+import { MessageHandlerService } from "../shared/message-handler/message-handler.service";
+
+@Component({
+  selector: "app-group",
+  templateUrl: "./group.component.html",
+  styleUrls: ["./group.component.scss"]
+})
+export class GroupComponent implements OnInit, OnDestroy {
+  searchTerm = "";
+  loading = true;
+  groups: UserGroup[] = [];
+  currentPage = 1;
+  totalCount = 0;
+  selectedGroups: UserGroup[] = [];
+  currentTerm = "";
+  delSub: Subscription;
+  batchOps = 'idle';
+  batchInfos = new Map();
+
+  @ViewChild(AddGroupModalComponent) newGroupModal: AddGroupModalComponent;
+
+  constructor(
+    private operationService: OperationService,
+    private translate: TranslateService,
+    private operateDialogService: ConfirmationDialogService,
+    private groupService: GroupService,
+    private msgHandler: MessageHandlerService,
+    private session: SessionService
+  ) {}
+
+  ngOnInit() {
+    this.loadData();
+    this.delSub = this.operateDialogService.confirmationConfirm$.subscribe(
+      message => {
+        if (
+          message &&
+          message.state === ConfirmationState.CONFIRMED &&
+          message.source === ConfirmationTargets.PROJECT_MEMBER
+        ) {
+          if (this.batchOps === 'delete') {
+            this.deleteGroups();
+          }
+        }
+      }
+    );
+  }
+  ngOnDestroy(): void {
+    this.delSub.unsubscribe();
+  }
+
+  refresh(): void {
+    this.loadData();
+  }
+
+  loadData(): void {
```

Comment:
I did not see any pagination or filter code in the loadData, where are they?

Reply:
because the group didn't have backend pagination. So I using datagrid's default front pagination.